cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go

```diff
@@ -78,6 +80,30 @@ func TestValidateAuthorizationModes(t *testing.T) {
 	}
 }
 
+func TestValidateNodeName(t *testing.T) {
+	hostname, _ := os.Hostname()
+	var tests = []struct {
+		s        string
+		f        *field.Path
+		expected bool
+	}{
+		{"", nil, false},                        // ok if not provided
+		{"1234", nil, true},                     // supported
+		{hostname, nil, true},                   // supported
```

Comment:
I think hard coding the value here makes sense instead of getting something from os.Hostname as that might depend on the env. What about if you run the unit test on a node with an upper-case hostname?